### ISSUE TITLE
1-Wire/Open Drain support

### ISSUE THREAD

**Issue by arlucio** [pulkin/micropython#20]
I'm trying to enable 1-Wire support on A9. For that it's necessary to implement Open-Drain GPIO Output somehow by software using SDK GPIO lib, doing something like this: https://embeddedartistry.com/blog/2018/06/11/simulating-open-drain-gpio-in-software/

From what I see, would be necessary to create a way to define a Pin in machine_pin as OpenDrain, to be able to call pin.init(pin.OPEN_DRAIN, pin.PULL_UP) on drivers/onewire/onewire.py. I've stumbled in how to do that, will try further but any help would be appreciated. 

ESP8266/ESP32/STM32 are some ports that already support 1-Wire/OpenDrain.

**Comment by huangguangw** [comment]
请问onewire在A9G上实现了吗？

**Comment by arlucio** [comment]
No, for this is still necessary to implement software Open-Drain for it.

**Comment by ahmadmicro** [comment]
any progress on this?
